lib/eco/incparser/error_recovery.py

```diff
@@ -36,6 +36,10 @@ def recover(self, error_node):
         # Get the character offset of the top node on the stack
         error_offset = self.stack_offset(len(self.stack))
 
+        # XXX When searching for an isolation node and we discover an iso node on
```

Comment:
Can we add an `assert` that catches the bad case, should it ever manifest again?